### ISSUE TITLE
From paper: Table 1, row 2 ('Shared' model - Mobilenetv2 backbone)

### ISSUE THREAD

**Issue by abhishekaich27** [brdav/bmtas#1]
Hi David,

I wanted to replicate the result in Table 1, row 2 (main paper), named "Shared". This seems to be composed of 1 shared encoder (MbNetv2) with _N_ task heads on the last layer of this encoder. Can you share the network implementation code (and possibly trained weights), only for this model?

I tried different variations on my end but couldn't replicate the result (e.g. am getting ~32 mIoU for SemSeg, whereas here we have 59.69 mIoU).

**Comment by brdav** [comment]
Hi,
the "Shared" model can be trained by feeding in a config file of all zeros, as follows:

`python main_branched.py --configuration </path/to/branch_config_with_all_zeros> --data_root . --tasks semseg,human_parts,sal,normals,edge`

As far as I can remember, I divided the learning rate by the number of tasks for each block, inspired by ASTMT. So to reproduce the "Shared" paper numbers, you should change the optimizer [here](https://github.com/brdav/bmtas/blob/b2d2ca06f370290ecbeaea0d21742edd0be54664/main_branched.py#L91) as follows:

```
optimizer = torch.optim.SGD([
         {'params': model.stem.parameters(), 'lr': 0.005 / len(tasks)},
         {'params': model.encoder.parameters(), 'lr': 0.005 / len(tasks)},
         {'params': model.decoder.parameters(), 'lr': 0.005},
     ], momentum=0.9, weight_decay=1e-4)
```

Hope that helps.
